python/lsst/analysis/tools/actions/plot/histPlot.py

```diff
@@ -129,28 +151,60 @@ def makePlot(
         # set up figure
         fig = plt.figure(dpi=300)
         hist_fig, side_fig = fig.subfigures(1, 2, wspace=0, width_ratios=[3, 1])
-        axs = self._makeAxes(hist_fig)
+        axs, ncols, nrows = self._makeAxes(hist_fig)
 
         # loop over each panel; plot histograms
-        cols = self._assignColors()
-        all_handles, all_nums, all_meds, all_mads = [], [], [], []
+        colors = self._assignColors()
+        nth_panel = len(self.panels)
+        nth_col = ncols
+        nth_row = nrows - 1
         for panel, ax in zip(self.panels, axs):
-            nums, meds, mads = self._makePanel(data, panel, ax, cols[panel], **kwargs)
+            nth_panel -= 1
+            nth_col = ncols - 1 if nth_col == 0 else nth_col - 1
+            if nth_panel == 0 and nrows * ncols - len(self.panels) > 0:
+                nth_col -= 1
+            # Set some font sizes based on number of panels being plotted.
+            label_font_size = max(6, 10 - nrows)
+            legend_font_size = max(4, int(8 - len(self.panels[panel].hists) / 2 - nrows // 2))  # type: ignore
```

Comment:
why do both `label_font_size` and `legend_font_size` need to be set inside this `for` loop? Can they be set once outside it?

Reply:
Ah, I missed the `self.panels[panel].hists` - that certainly does need to be assigned for each panel. Still not sure about `label_font_size` however.